src/selectors/post.js

```diff
@@ -25,9 +26,10 @@ export const selectPosts = state => state.json.get(POSTS, Immutable.Map())
 
 // Requires `postId`, `post` or `params.token` to be found in props
 export const selectPost = createSelector(
-  [selectPropsPostId, selectParamsToken, selectPosts], (id, token, posts) => {
+  [selectPropsPost, selectPropsPostId, selectParamsToken, selectPosts],
+  (propsPost, id, token, posts) => {
     if (id) {
-      return posts.get(id, Immutable.Map())
+      return posts.get(id, propsPost || Immutable.Map())
```

Comment:
This strays from the webapp a bit, but since we don't have the json state we need to pull the post from the props...